plugins/wasm-cpp/extensions/jwt_auth/README_EN.md

```diff
@@ -0,0 +1,399 @@
+
+#### 1.3.2 Composition of JWT
+
+`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ`
+
+As shown in the example above, JWT is a string consisting of three parts:
+
+- Header
+- Payload
+- Signature
+
+**Header**
+
+The header of the JWT carries two pieces of information:
+
+- The type of the token, which is JWT
+- The algorithm used for encryption
+
+The gateway supports the following encryption algorithms:
+
+```text
+ES256, ES384, ES512,
+HS256, HS384, HS512,
+RS256, RS384, RS512,
+PS256, PS384, PS512,
+EdDSA
+```
+
+The complete header looks like the following JSON:
+
+```js
+{
+  'typ': 'JWT',
+  'alg': 'HS256'
+}
+```
+
+The header is then Base64-encoded (this encoding can be symmetrically decoded), forming the first part:
+
+`eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9`
+
+**Payload**
+
+The payload is where the actual information is stored. The details are defined below:
+
+```text
+iss: The issuer of the token. This indicates who created the token and is a string value.
+sub: The subject identifier. This is the unique identifier for the end user provided by the issuer, and is no longer than 255 ASCII characters, and is case-sensitive within the issuer's scope.
+aud: The audience(s) of the token, which is an array of case-sensitive strings.
+exp: The expiration time of the token, which is a Unix timestamp in seconds, and any token that is issued after that time is invalid.
+iat: The time at which the token was issued, also a Unix timestamp in seconds.
+jti: The unique identifier for the token, and the value is unique for every token created by the issuer. It is usually a cryptographically random value to prevent conflicts. This component adds a random entropy that an attacker cannot obtain to the structured token, making it more difficult for the token to be guessed or replayed.
+```
+
+Custom fields for a user system can also be added, such as the example below adding a "name" field for the user's nickname:
+
+```js
+{
+  "sub": "1234567890",
+  "name": "John Doe"
+}
+```
+
+The payload is then Base64-encoded, forming the second part of the JWT:
+
+`JTdCJTBBJTIwJTIwJTIyc3ViJTIyJTNBJTIwJTIyMTIzNDU2Nzg5MCUyMiUyQyUwQSUyMCUyMCUyMm5hbWUlMjIlM0ElMjAlMjJKb2huJTIwRG9lJTIyJTBBJTdE`
+
+**Signature**
+
+This part is a string that consists of the Base64-encoded header and Base64-encoded payload concatenated with a period, followed by the encryption of the resulting string using the algorithm specified in the header (where $secret represents the user's private key).
+
+```js
+var encodedString = base64UrlEncode(header) + '.' + base64UrlEncode(payload);
+var signature = HMACSHA256(encodedString, '$secret');
+```
+
+These three parts are then concatenated using periods to form the complete JWT string as shown in the example in section 1.3.2.
```

Comment:
It sounds strange because we are still in section 1.3.2 here. The original text says "就构成了 1.3.2 节最开始的JWT示例". Do you think the meaning of "最开始" is mandatory here?